app/controllers/schools/advice/electricity_long_term_controller.rb

```diff
@@ -5,13 +5,68 @@ def insights
       end
 
       def analysis
+        @analysis_dates = analysis_dates
+
+        @annual_usage = usage_service.annual_usage
+        @vs_benchmark = usage_service.annual_usage_vs_benchmark(compare: :benchmark_school)
+        @vs_exemplar = usage_service.annual_usage_vs_benchmark(compare: :exemplar_school)
+
+        @estimated_savings_vs_exemplar = usage_service.estimated_savings(versus: :exemplar_school)
+        @estimated_savings_vs_benchmark = usage_service.estimated_savings(versus: :benchmark_school)
+
+        @multiple_meters = @school.meters.electricity.count > 1
+        if @multiple_meters
+          @annual_usage_meter_breakdown = usage_service.annual_usage_meter_breakdown
+          @meters_for_breakdown = sorted_meters_for_breakdown(@annual_usage_meter_breakdown)
+        end
       end
 
       private
 
       def advice_page_key
         :electricity_long_term
       end
+
+      def create_analysable
+        usage_service
+      end
+
+      def sorted_meters_for_breakdown(annual_usage_meter_breakdown)
+        meters = {}
+        annual_usage_meter_breakdown.meters.each do |mpan_mprn|
+          meters[mpan_mprn] = @school.meters.find_by_mpan_mprn(mpan_mprn)
+        end
+        meters.sort { |a, b| a[1].name_or_mpan_mprn <=> b[1].name_or_mpan_mprn }
```

Comment:
can the lookup and sort here can be done in the db in a single query?  Something like...
```
def sorted_meters_for_breakdown(annual_usage_meter_breakdown)
  meters = @school.meters.where(mpan_mprn: annual_usage_meter_breakdown.meters).order(:name, :mpan_mprn)
  meters.each_with_object({}) { |meter, meters| meters[meter.mpan_mprn] = meter }
end
```